src/main/scala/com/codacy/client/stash/client/StashClient.scala

```diff
@@ -55,105 +41,101 @@ class StashClient(baseUrl: String, authenticator: Option[Authenticator] = None,
     }
   }
 
-  def postForm[T](request: Request[T], values: Map[String, Seq[String]])(implicit reader: Reads[T]): RequestResponse[T] = {
-    performRequest("POST", request, values)
-  }
-
   def postJson[T](request: Request[T], values: JsValue)(implicit reader: Reads[T]): RequestResponse[T] = {
     performRequest("POST", request, values)
   }
 
-  def putForm[T](request: Request[T], values: Map[String, Seq[String]])(implicit reader: Reads[T]): RequestResponse[T] = {
-    performRequest("PUT", request, values)
-  }
-
   def putJson[T](request: Request[T], values: JsValue)(implicit reader: Reads[T]): RequestResponse[T] = {
     performRequest("PUT", request, values)
   }
 
-  // Wrap request with authentication options
-  private def withAuthentication(request: WSRequest): WSRequest = authenticator match {
-    case Some(auth) => auth.withAuthentication(request)
-    case None => request
-  }
+  //   Wrap request with authentication options
+  private def withAuthentication(request: HttpRequest): HttpRequest =
+    authenticator match {
+      case Some(auth) => auth.withAuthentication(request)
+      case None => request
+    }
 
   /*
    * Does an API request
    */
-  private def performRequest[D, T](method: String, request: Request[T], values: D)
-                                  (implicit reader: Reads[T], writer: Writeable[D], contentType: ContentTypeOf[D]): RequestResponse[T] = {
-    val url = generateUrl(request.url)
-
-    withClient { client =>
-      val jpromise = withAuthentication(
-        client
-          .url(url)
-          .withFollowRedirects(follow = true)
-          .withMethod(method)
-          .withBody(values)
-      ).execute()
-      val result = Await.result(jpromise, requestTimeout)
-
-      Try(result.body).map {
-        case body if Seq(HTTPStatusCodes.OK, HTTPStatusCodes.CREATED).contains(result.status) =>
-          parseJson[T](body) match {
-            case Right(json) => valueOrError[T](json)
-            case Left(response) => response
-          }
-        case _ if result.status == HTTPStatusCodes.NO_CONTENT =>
-          RequestResponse[T](None)
-        case body =>
-          getError[T](result.status, result.statusText, body)
-      }
-        .getOrElse(getError[T](result.status, result.statusText))
+  private def performRequest[T](method: String, request: Request[T], values: JsValue)(
+      implicit reader: Reads[T]
+  ): RequestResponse[T] = {
+    doRequest[T](request.url, method, Option(values)) match {
+      case Right((200 /*HTTPStatusCodes.OK*/ | 201 /*HTTPStatusCodes.CREATED*/, body)) =>
```

Comment:
put the numbers as values with a  good name in the companion object instead of relying on comments